FuelLabs/fuel-indexer
# Implement contract subscriptions

### Description
- This PR:
  - Removes unused manifest fields 
    - `test_events`
    - `handlers`
  - Implements `start_block` and `contract_id` functionality
 - Adds separate assets for macro testing
 
### Testing steps
- Integration tests tracked in https://github.com/FuelLabs/fuel-indexer/issues/272

- [ ] Uncomment out `contract_id` and `start_block` in `fuel-indexer-tests/assets/fuel_indexer_test.yaml` 
- [ ] View the generated code `cd cd fuel-indexer-tests/components/indices/fuel-indexer-test && cargo rustc --profile=check -- -Zunpretty=expanded`
  - You should see the start block conditional as well as the contract ID conditional in `handle_events`'s function body
